iceoryx_posh/doc/chunk_header.md

```diff
@@ -0,0 +1,218 @@
+# Summary
+
+Chunks are the transport capsules in iceoryx. They store data from a publisher as payload and are sent to one or more subscriber. Furthermore, there is some meta information which is stored alongside the payload, e.g. the size and the origin of the chunk. This data is composed in the `ChunkHeader` and located at the front of the chunk. Custom meta information can be added to extend the data from the `ChunkHeader` and tailor the chunk to specific use cases. While this makes the chunk layout more complex, this complexity would otherwise be distributed at different locations all over the code base, e.g. in the request/response feature. Additionally, the adjustments for the custom header makes arbitrary alignments of the payload trivial to implement.
+
+# Terminology
+
+| Name              | Description                                              |
+| :---------------- | :------------------------------------------------------- |
+| Chunk             | a piece of memory                                        |
+| Chunk Header      | contains meta information related to the chunk           |
+| Custom Header     | contains custom meta information, e.g. timestamps        |
+| Payload           | the user data                                            |
+| Back-Offset       | offset stored in front of the payload to calculate back to the chunk header |
+
+# Design
+
+## Considerations
+
+- it's not uncommon to record chunks for a later replay -> detect incompatibilities on replay
+- iceoryx runs on multiple platforms -> endianness of recorded chunks might differ
+- for tracing, a chunk should be uniquely identifiable -> store origin and sequence number
+- the chunk is located in the shared memory, which will be mapped to arbitrary positions in the address space of various processes -> no absolute pointer are allowed
+- aligning the `ChunkHeader` to 32 bytes will ensure that all member are on the same cache line and will improve performance
+- in order to reduce complexity, the alignment of the custom header must not exceed the alignment of the `ChunkHeader`
+
+## Solution
+
+### ChunkHeader Definition
+```
+struct alignas(32) ChunkHeader
+{
+    uint32_t m_chunkSize;
+    uint8_t m_chunkHeaderVersion;
+    uint8_t m_reserved1;
+    uint8_t m_reserved2;
+    uint8_t m_reserved3;
+    uint64_t m_originId;
+    uint64_t m_sequenceNumber;
+    uint32_t m_payloadSize;
+    uint32_t m_payloadOffset;
+};
+```
+
+- **m_chunkSize** is the size of the whole chunk
+- **m_chunkHeaderVersion** is used to detect incompatibilities for record&replay functionality
```

Comment:
Couldn't this be a compile time thing? How are we dealing with this currently? Using `iceoryx_posh/include/iceoryx_posh/version/version_info.hpp`?

Reply:
yes, it would be ca compile time thing, so this should probably be a `const uint8_t m_chunkHeaderVersion`. Have to think about the implications about this, e.g. the `ChunkHeader` wouldn't be copy asignable anymore